### ISSUE TITLE
New functionality -> set a label with face number while multi-face recognition

### ISSUE THREAD

**Issue by Macleopard** [cosanlab/py-feat#198]
![image](https://github.com/cosanlab/py-feat/assets/9639975/6fb9bee3-b7b3-486c-9ac5-edb77090d936)
Hello, in example, I have a photo with two faces. Faces are highlighted in the image, but it is difficult to understand for which face certain emotions are defined. Please, add label for this

**Comment by juaninachon** [comment]
Hi! I'm running a _detector.detect_video_ with:

_Detector(face_model="retinaface",
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; landmark_model="mobilefacenet",
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  au_model="xgb",
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; emotion_model="resmasknet",
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; facepose_model="img2pose",
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; **identity_model="facenet"**,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; device="cuda")_

Is there a column in the _detector.to_csv_  that works as an identity label? _Face Rect_ coordinates are not very useful if subjects move around.

![image](https://github.com/cosanlab/py-feat/assets/51125660/4930b087-4d01-4ed4-88ee-d774a287d86a)

Loving pyfeat!

**Comment by Macleopard** [comment]
> Hi! I'm running a _detector.detect_video_ with:
> 
> _Detector(face_model="retinaface",               landmark_model="mobilefacenet",               au_model="xgb",               emotion_model="resmasknet",               facepose_model="img2pose",               **identity_model="facenet"**,               device="cuda")_
> 
> Is there a column in the _detector.to_csv_ that works as an identity label? _Face Rect_ coordinates are not very useful if subjects move around.
> 
> ![image](https://private-user-images.githubusercontent.com/51125660/291315047-4930b087-4d01-4ed4-88ee-d774a287d86a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5MTY0NTMsIm5iZiI6MTcwMjkxNjE1MywicGF0aCI6Ii81MTEyNTY2MC8yOTEzMTUwNDctNDkzMGIwODctNGQwMS00ZWQ0LTg4ZWUtZDc3NGEyODdkODZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDE2MTU1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0NWI4OTQyZjI0MTVlZGY3NmJkZTU1MThiYjM1MjZmMDA5M2RjYWU5ODQwNTVjYzcxNmI0Y2ZlMTVmNTRmZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cX768jLnfP-bzDmF-wg_x1OLABmdxedwSYWLRvSclyI)
> 
> Loving pyfeat!

No, I recognize emotions in the photo, and I need to understand on the final image who owns what emotions, sometimes from the result of PyFeat it is not clear who is feeling what emotions

**Comment by ljchang** [comment]
We have recently added a new model to embed faces into a latent representation that is commonly used in identity tracking. We have added a simple clustering algorithm to try to figure out which faces are the same across frames that you might want to try playing with. However, it's important to note that this is definitely a work in progress and we haven't added this to our plots yet. Checkout PR #180 and #181. We still need to write a tutorial on how to use it, which should be in our next major release. (We are also refactoring our plots to be more aesthetically pleasing and interactive).

**Comment by juaninachon** [comment]
Thank you! I'll check it out as soon as I'm back in the lab. Looking
forward
to that upcoming release!

Kind regards,

Juan

**Comment by ljchang** [comment]
Added: in multi-face frames, plot_detections now labels each facebox with its face number (1-based, detection order) and its identity when available, so multi-face plots are readable. (on v0.7-dev; ships in v0.7.0).
